src/bp/ui-shared/src/TreeView/index.tsx

```diff
@@ -86,15 +98,15 @@ const TreeView = <T extends {}>(props: TreeViewProps<T>) => {
 
   const handleNodeDoubleClick = (selectedNode: TreeNode<T>) => {
     if (selectedNode.nodeData) {
-      props.onDoubleClick?.(selectedNode.nodeData, 'document')
+      props.onDoubleClick?.(selectedNode.nodeData, selectedNode.nodeData?.type || 'document')
```

Comment:
You already have access to the whole nodeData object, so you can check the type in onDoubleClick in the other component. We should keep 'document' for consistency